src/main/java/org/opensearch/neuralsearch/processor/NeuralQueryProcessor.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.util.Map;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.neuralsearch.query.visitor.NeuralSearchQueryVisitor;
+import org.opensearch.search.pipeline.AbstractProcessor;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchRequestProcessor;
+
+/**
+ * Neural Search Query Request Processor
+ */
+public class NeuralQueryProcessor extends AbstractProcessor implements SearchRequestProcessor {
+
+    /**
+     * Key to reference this processor type from a search pipeline.
+     */
+    public static final String TYPE = "neural_query";
+
+    final String modelId;
+
+    final Map<String, Object> neuralFieldDefaultIdMap;
+
+    /**
+     * Returns the type of the processor.
+     *
+     * @return The processor type.
+     */
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    protected NeuralQueryProcessor(
+        String tag,
+        String description,
+        boolean ignoreFailure,
+        String modelId,
+        Map<String, Object> neuralFieldDefaultIdMap
+    ) {
+        super(tag, description, ignoreFailure);
+        this.modelId = modelId;
+        this.neuralFieldDefaultIdMap = neuralFieldDefaultIdMap;
+    }
+
+    /**
+     * Processes the Search Request.
+     *
+     * @return The Search Request.
+     */
+    @Override
+    public SearchRequest processRequest(SearchRequest searchRequest) {
+        QueryBuilder queryBuilder = searchRequest.source().query();
+        queryBuilder.visit(new NeuralSearchQueryVisitor(modelId, neuralFieldDefaultIdMap));
+        return searchRequest;
+    }
+
+    public static class Factory implements Processor.Factory<SearchRequestProcessor> {
```

Comment:
this should be a separate class, please check existing factory classes for text_embedding for example

Reply:
Text embedding processor is a ingest processor. Neural Query Processor is a search request processor. I have followed the same template of search request processor how they are implemented in OpenSearch. We should follow the same template of OS

[FilterQueryRequestProcessor](https://github.com/opensearch-project/OpenSearch/blob/main/modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/FilterQueryRequestProcessor.java)

Reply:
I don't see a reason why to follow that pattern, what makes more sense to me for decision is if that class has important and complex logic/responsibility, or not. We've done a separate class for [normalization processor factory](https://github.com/opensearch-project/neural-search/blob/main/src/main/java/org/opensearch/neuralsearch/processor/factory/NormalizationProcessorFactory.java), processor is a search processor. 
If you think factory for your processor is simple we can leave it as inner class

Reply:
Yes it is simple.